exp/addons/internal/controllers/clusterresourceset_controller_test.go

```diff
@@ -990,10 +1010,19 @@ metadata:
 		g.Expect(env.Patch(ctx, clusterResourceSetInstance, client.MergeFrom(clusterResourceSetInstance.DeepCopy()))).To(Succeed())
 
 		// Wait until ClusterResourceSetBinding is created for the Cluster
-		g.Eventually(func(g Gomega) {
+		g.Eventually(func() bool {
+			// the binding must exists and track resource being applied
 			binding := &addonsv1.ClusterResourceSetBinding{}
-			g.Expect(env.Get(ctx, clusterResourceSetBindingKey, binding)).Should(Succeed())
-		}, timeout).Should(Succeed())
+			if err := env.Get(ctx, clusterResourceSetBindingKey, binding); err != nil {
+				return false
+			}
+			for _, b := range binding.Spec.Bindings {
+				if len(b.Resources) == 0 {
+					return false
+				}
+			}
+			return true
```

Comment:
Not sure this works as expected. This will return true if the Bindings array is empty?

Reply:
Can we add this before the for loop?
```go
			if len(binding.Spec.Bindings) == 0{
				return false
			}
```

I know with the *current* implementation your check should also catch it (because we append to bindings via defer patch when the svc doesn't exist), but seems better to me to verify it without relying on details of the implementation

Reply:
done!